Sources/_StringProcessing/RegexDSL/DSL.swift

```diff
@@ -59,16 +69,23 @@ extension CharacterClass: RegexProtocol {
 
 // MARK: Repetition
 
+private func quant(
```

Comment:
IMO using the enum constructor `.quantification(...)` would be simpler and cleaner than a top-level function.